src/python/pants/backend/python/target_types_test.py

```diff
@@ -364,35 +366,90 @@ def test_parse_requirements_file() -> None:
     }
 
 
-def test_python_distribution_dependency_injection() -> None:
+def test_inject_python_distribution_dependencies() -> None:
     rule_runner = RuleRunner(
         rules=[
             inject_python_distribution_dependencies,
+            resolve_pex_entry_point,
+            resolve_python_distribution_entry_points,
+            *import_rules(),
+            *python_sources.rules(),
             QueryRule(InjectedDependencies, [InjectPythonDistributionDependencies]),
         ],
-        target_types=[PythonDistribution, PexBinary],
+        target_types=[PythonDistribution, PythonRequirementLibrary, PythonLibrary, PexBinary],
         objects={"setup_py": PythonArtifact},
     )
+    rule_runner.add_to_build_file(
+        "",
+        dedent(
+            """\
+            python_requirement_library(
+                name='ansicolors',
+                requirements=['ansicolors'],
+                module_mapping={'ansicolors': ['colors']},
+            )
+            """
+        ),
+    )
+    rule_runner.create_file("project/app.py")
     rule_runner.add_to_build_file(
         "project",
         dedent(
             """\
-            pex_binary(name="my_binary")
+            pex_binary(name="my_binary", entry_point="who_knows.module:main")
+
+            python_library(name="my_library", sources=["app.py"])
+
             python_distribution(
-                name="dist",
+                name="dist-a",
                 provides=setup_py(
-                    name='my-dist'
+                    name='my-dist-a'
                 ).with_binaries({"my_cmd": ":my_binary"})
             )
+
+            python_distribution(
+                name="dist-b",
+                provides=setup_py(
+                    name="my-dist-b"
+                ),
+                entry_points={
+                    "console_scripts":{
+                        "b_cmd": "project.app:main",
+                        "cmd_2": "//project:my_binary",
+                    }
+                },
+            )
             """
         ),
     )
-    tgt = rule_runner.get_target(Address("project", target_name="dist"))
-    injected = rule_runner.request(
-        InjectedDependencies,
-        [InjectPythonDistributionDependencies(tgt[PythonDistributionDependencies])],
+    rule_runner.create_file("who_knows/module.py")
+    rule_runner.add_to_build_file(
+        "who_knows",
+        dedent(
+            """\
+            python_library(name="random_lib", sources=["module.py"])
+            """
+        ),
+    )
+
+    def assert_injected(address: Address, *expected: Address) -> None:
```

Comment:
Imo, more clear is to use `list[Address]` for expected. I think it better expresses that the first arg is different than the rest.

Reply:
ok. I was contemplating that...